backend/src/main/java/com/google/sps/servlets/SearchMatchServlet.java

```diff
@@ -142,9 +153,24 @@ private void sendMatchResponse(HttpServletResponse response, Match match) throws
     matchExists.put(JSON_FIRST_PARTICIPANT_USERNAME, match.getFirstParticipantUsername());
     matchExists.put(JSON_SECOND_PARTICIPANT_USERNAME, match.getSecondParticipantUsername());
     matchExists.put(JSON_DURATION, match.getDuration());
-
+    try {
+      emailNotifier.sendMatchEmail(getUsername(), userService.getCurrentUser().getEmail());
+    } catch (MessagingException e) {
+      e.printStackTrace();
+    }
     // Send the JSON back as the response
     response.setContentType("application/json");
     response.getWriter().println(matchExists.toString());
   }
+
+  /** Retrieve user email address via Users API and parse for username */
+  private String getUsername() {
+    UserService userService = UserServiceFactory.getUserService();
+    String email = userService.getCurrentUser().getEmail();
+    return email != null ? email.split("@")[0] : null;
```

Comment:
when will email be null?

Reply:
The user doesn't exists

Reply:
Can that happen normally?

Reply:
I'm not too familiar with the backend but I don't believe so as an error will be thrown if such field is null form what i've seen

Reply:
This will only be null if the Users API returns null (can't retrieve email). Since I do not think this can normally happen, I think we can just return the email the Users API returns and move forward from there. Otherwise, if we choose to keep the null return value from this method, what do you think about adding a null check to the places where this method is called elsewhere in the servlet?

Reply:
Adding null checking sounds like it could be fine !